controllers/opentelemetrycollector_controller_test.go

```diff
@@ -17,240 +17,24 @@ package controllers_test
 import (
 	"context"
 	"errors"
-	"fmt"
 	"testing"
 
-	routev1 "github.com/openshift/api/route/v1"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-	appsv1 "k8s.io/api/apps/v1"
-	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
-	"k8s.io/client-go/tools/record"
 	"k8s.io/kubectl/pkg/scheme"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 	k8sconfig "sigs.k8s.io/controller-runtime/pkg/client/config"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 	k8sreconcile "sigs.k8s.io/controller-runtime/pkg/reconcile"
 
-	"github.com/open-telemetry/opentelemetry-operator/apis/v1alpha1"
 	"github.com/open-telemetry/opentelemetry-operator/controllers"
 	"github.com/open-telemetry/opentelemetry-operator/internal/config"
 	"github.com/open-telemetry/opentelemetry-operator/internal/manifests"
 	"github.com/open-telemetry/opentelemetry-operator/pkg/autodetect"
 )
 
 var logger = logf.Log.WithName("unit-tests")
-var mockAutoDetector = &mockAutoDetect{
-	OpenShiftRoutesAvailabilityFunc: func() (autodetect.OpenShiftRoutesAvailability, error) {
-		return autodetect.OpenShiftRoutesAvailable, nil
-	},
-}
-
-func TestNewObjectsOnReconciliation(t *testing.T) {
```

Comment:
this delete is okay IMO, when we went through this together I realized that all of these tests were actually superseded by the tests in reconcile_test.go

Reply:
However, it looks scary :cow2:

Reply:
Yeah, these were particularly bad tests because they actually had no config as part of them, it only checked that things like ingress resources would be generated based on the config. Now that we have enhanced checks for valid configuration all of these tests would fail. We could fix these by supplying real config, BUT all of these checks are handled in the reconcile_test.go file and builder_test.go which check for the existence of expected resource.